src/js/angular/core/directives/yasgui-component/yasgui-component.directive.js

```diff
@@ -0,0 +1,624 @@
+                    status: {
+                        processedEntities: 0,
+                        estimatedEntities: 0,
+                        indexedEntities: 0,
+                        entitiesPerSecond: 0
+                    },
+                    actionName,
+                    eta: "-",
+                    inline: false,
+                    iri,
+                    name: connectorName,
+                    doneCallback: function () {
+                        connectorProgressModal.dismiss('cancel');
+                    }
+                };
+                progressScope.getHumanReadableSeconds = $scope.getHumanReadableSeconds;
+
+                const connectorProgressModal = $uibModal.open({
+                    templateUrl: 'pages/connectorProgress.html',
+                    controller: 'CreateProgressCtrl',
+                    size: 'lg',
+                    backdrop: 'static',
+                    scope: progressScope
+                });
+                return connectorProgressModal;
+            };
+
+            const toCreateCommandResponse = (response, tabId) => {
+                const connectorProgressModal = createConnectorProgressDialog($translate.instant('externalsync.creating'), response.data.iri, response.data.name);
+                tabIdToConnectorProgressModalMapping.set(tabId, connectorProgressModal);
+                return new BeforeUpdateQueryResult(BeforeUpdateQueryResultStatus.SUCCESS, 'created.connector', getCommandParameters(response));
+            };
+
+            const toRepairCommandResponse = (response, tabId) => {
+                const connectorProgressModal = createConnectorProgressDialog($translate.instant('externalsync.repairing'), response.data.iri, response.data.name);
+                tabIdToConnectorProgressModalMapping.set(tabId, connectorProgressModal);
+                return new BeforeUpdateQueryResult(BeforeUpdateQueryResultStatus.SUCCESS, 'query.editor.repaired.connector', getCommandParameters(response));
+            };
+
+            const toDropCommandResponse = (response) => {
+                return new BeforeUpdateQueryResult(BeforeUpdateQueryResultStatus.SUCCESS, 'externalsync.delete.success.msg', getCommandParameters(response));
+            };
+
+            const getBeforeUpdateQueryHandler = () => (query, tabId) => {
+                return new Promise(function (resolve) {
+                    ConnectorsRestService.checkConnector(query)
+                        .then((response) => {
+                            if (!response.data.command) {
+                                resolve(toNoCommandResponse());
+                                return;
+                            }
+                            if (!response.data.hasSupport) {
+                                resolve(toHasNotSupport(response));
+                                return;
+                            }
+                            switch (response.data.command) {
```

Comment:
Can you stick withh the `if-else` please? Using both `if-else` and `switch-case` is kind of confusing for reading.

Reply:
Done